src/client/java/net/neoforged/neoforge/client/gui/widget/ScrollPanel.java

```diff
@@ -9,24 +9,18 @@
 import java.util.List;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.gui.GuiGraphics;
-import net.minecraft.client.gui.components.Renderable;
-import net.minecraft.client.gui.components.events.AbstractContainerEventHandler;
+import net.minecraft.client.gui.components.AbstractContainerWidget;
 import net.minecraft.client.gui.components.events.GuiEventListener;
-import net.minecraft.client.gui.narration.NarratableEntry;
+import net.minecraft.client.gui.navigation.ScreenRectangle;
 import net.minecraft.client.gui.screens.Screen;
 import net.minecraft.client.input.MouseButtonEvent;
+import net.minecraft.network.chat.Component;
 
 /**
  * Abstract scroll panel class.
  */
-public abstract class ScrollPanel extends AbstractContainerEventHandler implements Renderable, NarratableEntry {
+public abstract class ScrollPanel extends AbstractContainerWidget {
     private final Minecraft client;
-    protected final int width;
-    protected final int height;
-    protected final int top;
-    protected final int bottom;
-    protected final int right;
-    protected final int left;
     private boolean scrolling;
     protected float scrollDistance;
```

Comment:
With the addition of the container widget, these fields are in its superclass. It should be updated to set and use those values instead.